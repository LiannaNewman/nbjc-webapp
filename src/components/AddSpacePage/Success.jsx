import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';

const styles = () => ({
  pageTitle: {
    marginTop: 60,
    marginBottom: 28,
  },
  check: {
    width: 63,
    height: 63,
    margin: '0 auto',
  },
  body: {
    marginTop: 28,
    marginBottom: 60,
  },
  buttonWrapper: {
    display: 'flex',
    justifyContent: 'center',
  },
});

const Success = ({ classes }) => (
  <>
    <Typography variant="h4" align="center" className={classes.pageTitle}>
      Space Submitted
    </Typography>
    <div className={classes.check}>
      <CheckCircleIcon color="primary" className={classes.check} />
    </div>
    <Typography variant="subtitle1" align="center" className={classes.body}>
      Thank you for adding a space to OurGuide. An administrator will review
      your submission. Check back soon to see your space in the application.
    </Typography>
    <div className={classes.buttonWrapper}>
      <Button
        variant="contained"
        color="secondary"
        align="center"
        fullWidth={!useMediaQuery('(min-width:376px)')}
        href="/search"
        disableElevation
      >
        Search for a space
      </Button>
    </div>
  </>
);

Success.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

Success.defaultProps = {};
export default withStyles(styles)(Success);
