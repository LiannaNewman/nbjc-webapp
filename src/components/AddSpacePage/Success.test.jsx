import React from 'react';
import { render } from '@testing-library/react';

import Success from './Success';

describe('Success', () => {
  test('renders', () => {
    const { asFragment } = render(
      <Success />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
