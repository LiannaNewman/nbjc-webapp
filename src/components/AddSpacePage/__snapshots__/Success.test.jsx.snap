// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success renders 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root Success-pageTitle-1 MuiTypography-h4 MuiTypography-alignCenter"
  >
    Space Submitted
  </h4>
  <div
    class="Success-check-2"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root Success-check-2 MuiSvgIcon-colorPrimary"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </div>
  <h6
    class="MuiTypography-root Success-body-3 MuiTypography-subtitle1 MuiTypography-alignCenter"
  >
    Thank you for adding a space to Lavender Book. An administrator will review your submission. Check back soon to see your space in the application.
  </h6>
  <div
    class="Success-buttonWrapper-4"
  >
    <a
      align="center"
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      href="/search"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Search for a space
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
